src/logging/writers/ascii/Ascii.cc

```diff
@@ -448,6 +457,11 @@ bool Ascii::DoInit(const WriterInfo& info, int num_fields, const threading::Fiel
 			ext += gzip_file_extension.empty() ? "gz" : gzip_file_extension;
 			}
 
+		if ( ! logdir.empty() )
+			{
+			fname = logdir.empty() ? fname : logdir + "/" + fname;
+			}
+
```

Comment:
What happens when the name given here already is a (or contains a) path? It might be worth checking that the filename does not already start with a slash.

Reply:
Multiple slashes are allowed, and way back to POSIX it seems 
Try for yourself 
user@Projects:test$ mkdir logdir
user@Projects:test$ mkdir logdir//test2
user@Projects:test$ mkdir logdir//test2//////test3
user@Projects:test$ touch file1
user@Projects:test$ touch logdir/file1
user@Projects:test$ touch logdir////file1
user@Projects:test$ touch logdir////test2//file2

Reply:
and fname is our "own", like dns.log - why would they contain a slash, since they are usually/currently written into the current directory.